### PR TITLE
The ID of the a patched or updated image is written to cache file.

### LINKED ISSUE #54: Add an updated SUSE image in the cache as a SUSE image
When creating a new updated image from the `update` command, the newly created image should be added to the cache as a SUSE image.

### PR DESCRIPTION
**Pull request by jloehel**
Fixes #54

Signed-off-by: Jürgen Löhel <jloehel@suse.com>

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -321,18 +321,18 @@ func commitContainerToImage(containerId, repo, tag, comment, author string) erro
 // The name of the new image is specified via target_repo and target_tag.
 // The container is always deleted.
 // If something goes wrong an error message is returned.
-func runCommandAndCommitToImage(img, target_repo, target_tag, cmd, comment, author string) error {
+func runCommandAndCommitToImage(img, target_repo, target_tag, cmd, comment, author string) (string, error) {
 	containerId, err := runCommandInContainer(img, []string{cmd}, true)
-	if err != nil {
-		return err
+	if err != nil || containerId == ""{
```
Comment (flavio) [inline client.go]: Is it really possible to have `err == nil` and `containerId == ""`? If that really happens I think we should return a proper error instead of a nil one (that would screw up checks).

File: client_test.go [hunk 2]
```diff
@@ -284,7 +284,7 @@ func TestRunCommandAndCommitToImageSuccess(t *testing.T) {
 	var err error
 
 	capture.All(func() {
-		err = runCommandAndCommitToImage(
+		imageId, err = runCommandAndCommitToImage(
```
Comment (flavio) [inline client_test.go]: `imageId` is not being used, replace that with `_`.

**Comment by flavio** [comment]
The go tests are failing, please fix them. (Just ignore the integration tests, I'm going to fix them)

File: client.go (hunk 1 above)
Comment (mssola) [inline client.go]: If `containerId == ""`, it's guaranteed that err will not be nil. Since the error should be descriptive enough, I'd say to keep it as it was.

File: client.go [hunk 3]
```diff
@@ -321,18 +321,18 @@ func commitContainerToImage(containerId, repo, tag, comment, author string) erro
 // The name of the new image is specified via target_repo and target_tag.
 // The container is always deleted.
 // If something goes wrong an error message is returned.
-func runCommandAndCommitToImage(img, target_repo, target_tag, cmd, comment, author string) error {
+func runCommandAndCommitToImage(img, target_repo, target_tag, cmd, comment, author string) (string, error) {
```
Comment (mssola) [inline client.go]: Update the documentation

File: client.go [hunk 4]
```diff
@@ -310,10 +310,10 @@ func removeContainer(id string) {
 // Create a Docker image from the container specified by containerId
 // repo is the name of the reposity, hence it includes also the namespace (eg: suse/sle11sp3)
 // tag is the version of the image (eg: 1.0.0)
-func commitContainerToImage(containerId, repo, tag, comment, author string) error {
+func commitContainerToImage(containerId, repo, tag, comment, author string) (string, error) {
```
Comment (mssola) [inline client.go]: Update the documentation if you're changing the signature of a function.

File: cache.go [hunk 5]
```diff
@@ -104,19 +104,25 @@ func (cd *cachedData) reset() {
 	cd.flush()
 }
 
-// Add the ID of the image identified by the provided name to the list
-// of the outdated images.
-func (cd *cachedData) addImageToListOfOutdatedOnes(name string) error {
-	imageId, err := getImageId(name)
+// Update the Cachefile after an update.
+// The ID of the outdated image will be add to outdated Images and
```
Comment (mssola) [inline cache.go]: typo: "will be add" -> "will be added"

File: cache.go [hunk 6]
```diff
@@ -104,19 +104,25 @@ func (cd *cachedData) reset() {
...
+// the ID of the new image will be add to the SUSE Images.
```
Comment (mssola) [inline cache.go]: same here

**Comment by mssola** [comment]
Generally speaking it looks good to me. Just fix the tests and all the notes.

File: cache.go [hunk 7]
```diff
@@ -104,19 +104,22 @@ func (cd *cachedData) reset() {
...
+// the ID of the new image will be added to the SUSE Images.
+func (cd *cachedData) updateCacheAfterUpdate(outdatedImg, updatedImgId string) error {
+	outdatedImgId, err := getImageId(outdatedImg)
 	if err != nil {
 		return err
 	}
-
-	if !arrayIncludeString(cd.Outdated, imageId) {
-		cd.Outdated = append(cd.Outdated, imageId)
+	if !arrayIncludeString(cd.Outdated, outdatedImgId) {
```
Comment (mssola) [inline cache.go]: Shouldn't you check that updatedImgId is in the cd.Suse array as you do with outdated images? In fact, IIRC you were doing this checking I'm referring before updating this patch. Why did you remove it ?

File: cache.go [hunk 8]
```diff
@@ -104,19 +104,22 @@ func (cd *cachedData) reset() {
...
+		cd.Outdated = append(cd.Outdated, outdatedImgId)
+		cd.Suse = append(cd.Suse, updatedImgId)
 		cd.flush()
 	}
+
 	return nil
+
```
Comment (mssola) [inline cache.go]: remove extra line

File: cache.go (hunk 7 above)
Comment (jloehel) [inline cache.go]: I thought the new id is unique, so a second check is not necessary.

File: client.go [hunk 9]
```diff
@@ -321,18 +322,19 @@ func commitContainerToImage(containerId, repo, tag, comment, author string) erro
...
+// Returns the ID of the new image on success.
+func runCommandAndCommitToImage(img, target_repo, target_tag, cmd, comment, author string) (string, error) {
 	containerId, err := runCommandInContainer(img, []string{cmd}, true)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	err = commitContainerToImage(containerId, target_repo, target_tag, comment, author)
+	ImageId, err := commitContainerToImage(containerId, target_repo, target_tag, comment, author)
```
Comment (mssola) [inline client.go]: variable follow lowerCamelCase ;)

**Comment by mssola** [comment]
I'd like to see in the tests that both if's in `updateCacheAfterUpdate` are being executed.

**Comment by mssola** [comment]
LGTM. @flavio what do you think ?
